src/workerd/api/queue.c++

```diff
@@ -386,23 +382,29 @@ void QueueMessage::retry() {
     IoContext::current().logWarning(msg);
     return;
   }
-  result->explicitRetries.findOrCreate(id, [this]() { return kj::heapString(id); } );
+
+  result->retries.upsert(kj::heapString(id), {});
+  KJ_IF_SOME(opts, options) {
+    KJ_IF_SOME(secs, opts.delaySeconds) {
+      result->retries.upsert(kj::heapString(id), {.delaySeconds = secs});
+    }
+  }
 }
 
 void QueueMessage::ack() {
   if (result->ackAll) {
     return;
   }
 
-  if (result->retryAll) {
+  if (result->retryAll.value) {
     auto msg = kj::str(
         "Received a call to ack() on message ", id, " after retryAll() was already called. "
         "Calling ack() on a message after calling retryAll() has no effect.");
     IoContext::current().logWarning(msg);
     return;
   }
 
-  if (result->explicitRetries.contains(id)) {
+  if (result->retries.find(id) != kj::none) {
```

Comment:
Hm, I'm actually not 100% sure how this will behave if the `id` is found but it doesn't have a corresponding `delaySeconds` set. Will the `jsg::Optional` returned by `retries.find` be implicitly coerced into a `kj::none`, causing this to be false when we expect it to be true?

It's probably ok since it'd be odd if a `Maybe<Maybe<>>` with a none purely in the inner `Maybe` compared as equal to `none`, but this seems worth double checking if there isn't already a test that exercises this code.

Reply:
I will check this and get back to you later on.

Reply:
Thanks for confirming offline that this is ok 👍